service_load_balancing/launch_type_fargate.tf

```diff
@@ -0,0 +1,87 @@
+// NOTE: var.launch_type = "FARGATE" if you want use following
+
+resource "aws_ecs_cluster" "fargate" {
+  count = "${ var.launch_type == "FARGATE" ? 1 : 0 }"
+  name  = "${var.cluster_name}"
+}
+
+resource "aws_ecs_service" "fargate" {
+  count                              = "${ var.launch_type == "FARGATE" ? 1 : 0 }"
+
+  name                               = "${var.name}"
+  cluster                            = "${aws_ecs_cluster.fargate.id}"
+  launch_type                        = "${var.launch_type}"
+  task_definition                    = "${aws_ecs_task_definition.fargate.arn}"
+
+  # As below is can be running in a service during a deployment
+  desired_count                      = "${var.desired_count}"
+  deployment_maximum_percent         = "${var.deployment_maximum_percent}"
+  deployment_minimum_healthy_percent = "${var.deployment_minimum_healthy_percent}"
+
+  network_configuration {
+    assign_public_ip = true
+    subnets          = ["${var.subnet_ids}"]
+    security_groups  = ["${var.security_group_ids}"]
+  }
+
+  load_balancer {
+    target_group_arn = "${var.target_group_arn}"
+    container_name   = "${var.container_name}"
+    container_port   = "${var.container_port}"
+  }
+
+  lifecycle {
+    # INFO: In the future, we support that U can customize
+    #       https://github.com/hashicorp/terraform/issues/3116
+    ignore_changes = ["desired_count", "task_definition"]
```

Comment:
[IMO] Isn't it better adding `task_definition` after the PR is merged?

Reply:
On this module, currently no-support task_definition updates changing.

We are waiting for https://github.com/hashicorp/terraform/issues/3116
and not yet fix-up ;-(

You can change to container_definition & task parameters via
[ecs_deploy](https://github.com/silinternational/ecs-deploy), AWS CodePipeline, and ecs-cli, ...

but we'll fixup in the future 👍